pkg/operator/ceph/cluster/osd/topology/topology.go

```diff
@@ -148,3 +149,112 @@ func GetDefaultTopologyLabels() string {
 
 	return strings.Join(Labels, ",")
 }
+
+// CheckTopologyConflicts verifies that:
+// 1. No child domain (e.g. rack) has fewer distinct values than its immediate parent.
+// 2. No topology value is used under more than one label key.
+// 3. No label value lives under more than one parent label value.
+func CheckTopologyConflicts(nodes *[]corev1.Node) error {
+	// 1. Build our ordered list of topology labels (region -> zone -> datacenter -> …), dropping hostname.
+	allLabels := allKubernetesTopologyLabelsOrdered()
+	var hierarchy []string
+	for _, lbl := range allLabels {
```

Comment:
For the variable names, it's preferred to have full words instead of abbreviations that are hard to understand at a glance, unless they are well-known vars. I'll give some suggestions for var names, but would suggest you review all of them.
```suggestion
	for _, label := range allLabels {
```

Reply:
done